x/batching/keeper/abci.go

```diff
@@ -34,37 +36,49 @@ func (k Keeper) EndBlock(ctx sdk.Context) (err error) {
 		return err
 	}
 
-	err = k.SetBatch(ctx, batch)
-	if err != nil {
-		return err
-	}
-	err = k.IncrementCurrentBatchNum(ctx)
+	err = k.SetNewBatch(ctx, batch)
 	if err != nil {
 		return err
 	}
-
 	return nil
 }
 
 func (k Keeper) ConstructBatch(ctx sdk.Context) (types.Batch, error) {
-	curBatchNum, err := k.GetCurrentBatchNum(ctx)
+	// Note current will be "old" for this new batch.
+	oldBatchNum, err := k.GetCurrentBatchNum(ctx)
 	if err != nil {
 		return types.Batch{}, err
 	}
-	dataEntries, dataRootHex, err := k.ConstructDataResultTree(ctx)
+	newBatchNum := oldBatchNum + 1
+
+	dataEntries, dataRoot, err := k.ConstructDataResultTree(ctx)
 	if err != nil {
 		return types.Batch{}, err
 	}
-	valEntries, valRootHex, err := k.ConstructValidatorTree(ctx)
+	valEntries, valRoot, err := k.ConstructValidatorTree(ctx)
 	if err != nil {
 		return types.Batch{}, err
 	}
 
+	// Compute the batch ID, which is defined as
+	// keccak256(batch_number, block_height, validator_root, results_root).
```

Comment:
We're missing the `proving_metadata` or however we decide to name the field, or did we decide against it in the end?

Reply:
Just saw the other PR, I'm ok with adding it there.

Reply:
By the way, according to the specs `proving_metadata` is not included when we derive the ID.

Reply:
I think that's because this decision came after that spec. We should check with Mario